### DISCUSSION TITLE
Para conseguir um emprego preciso saber Java?

### DISCUSSION THREAD

**Discussion by matheusmat** [frontendbr/forum#1443] [Discussão]
Olá amigos e colegas, boa tarde!

Bem, sou um formando de computação e aqui onde estudo, desde do começo, quase tudo engloba Java, praticamente todas as cadeiras práticas e eu não gosto e/ou não sou confortável com desenvolvimento backend. Enfim, o problema é que o mercado aqui da região onde moro (nordeste) as vagas de estágio quase sempre é focado no desenvolvimento backend assim como as de emprego também, até quando vejo vagas para frontend sempre tem algo com jsp, jsf, thymeleaf no meio. Eu não teria problema em me mudar pra outro estado mas o que me incomoda bastante é que os professores e colegas são fanáticos por java, se por exemplo você não faz parte desse mundo automaticamente terá que aprender pra conseguir trabalho, que sem java você será mal sucedido e que "só" fazendo "telinha" não irá me fazer ter sucesso na carreira.

Eu gosto muito de ui design também e me viro bem com html, css, js, vue, sass, gulp, django e estudo react atualmente. Gostaria como realmente é o mercado, se eles estão certo e eu errado por está apostando em algo que não seja java, se puderem me ajudar, fica meu muito obrigado desde já.

**Comment by gabriel-hahn** [comment]
Olá Matheus,

Java é e sempre foi uma linguagem muito adotada pelas empresas no Brasil e no mundo. Atualmente também trabalho com Java no back e realmente a maioria das vagas pedem o conhecimento na linguagem. Acredito que é algo que muda um pouco de figura em cidades maiores, em que muitas empresas possuem o profissional Front-end bem distinguido do Back-end. Vejo muitas vagas específicas para Front pedindo somente o básico de alguma linguagem Back-end.

Aprender Java a fundo vai te abrir muitas portas e oportunidades de trabalho, porém existe muitas vagas em outras linguagens, além das específicas de Front-end. Dá uma curtida em vagas no Linkedin nesse perfil e continua se aperfeiçoando nas tecnologias que tu curte :)

**Comment by willycamargo** [comment]
Estou trabalhando há 6 anos e nunca escrevi uma linha de Java profissionalmente.

Não, você **não precisa** saber java para conseguir um emprego. Mas se o mercado local pede e você quer se enquadrar no mercado local, então já tem sua resposta.

O fato é que Java é uma linguagem estável e consolidada no mercado. Sempre vai acabar encontrando projetos desenvolvidos com Java ou tecnologias baseadas em Java. Então saber Java é um diferencial interessante mesmo quando seu foco não vai ser trabalhar exatamente com essa linguagem.

É interessante tentar olhar Java com bons olhos. Desenvolvedores de outras linguagens tendem a ter preconceito com a linguagem (eu ja tive muito), mas com um pouco de carinho conseguimos enxergar o potencial e importância do Java no mercado nacional (e também internacional, eu acho).
___________

Ah... Se quiser ter uma visão geral sobre o que o mercado espera de um desenvolvedor frontend, de uma olhada no repositório de vagas: https://github.com/frontendbr/vagas/issues


Abrs e boa sorte.
